src/internal/proc.js

```diff
@@ -151,6 +151,19 @@ function createTaskIterator({ context, fn, args }) {
       )
 }
 
+function getLocation(instrumented) {
+  return (
+    // TODO move symbol creation to redux-saga side
+    instrumented[Symbol.for('babel-plugin-transform-redux-saga-source')] || instrumented.__source || null
+  )
+}
+export function getMetaInfo(fn) {
+  return {
+    name: fn.name || 'anonymous',
+    location: getLocation(fn),
+  }
+}
+
 export default function proc(
```

Comment:
it seems to me that we could accept saga (generator) instead of iterator here and just use saga's static properties later when needed - no need for meta object as arguments, we could even (possibly) eliminate passing explicit `name` arg

Reply:
looking through the code now and im not sure if it can actually be done - but would certainly be worth checking

Reply:
I looked through the `proc` code and not sure that's possible without refactoring, so I'd prefer keep that out of scope of that PR